storage/graphdriver/rawblock/driver.go

```diff
@@ -0,0 +1,245 @@
+		return fmt.Errorf("Failed to create block:%v:%s", err, string(out))
+	}
+	if out, err := exec.Command("losetup", "-f", "--show", d.block(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to lopsetup the block:%v:%s", err, string(out))
+	} else {
+		defer exec.Command("losetup", "-d", string(out)).CombinedOutput()
+		if out1, err := exec.Command("mkfs", "-t", d.blockFs, d.block(id)).CombinedOutput(); err != nil {
+			return fmt.Errorf("Failed to mkfs the block:%v:%s", err, string(out1))
+		}
+		return nil
+	}
+
+}
+
+// Create prepares the filesystem for the rawblock driver and copies the block from the parent.
+func (d *Driver) Create(id, parent, mountLabel string) error {
+	rootUID, rootGID, err := idtools.GetRootUIDGID(d.uidMaps, d.gidMaps)
+	if err != nil {
+		return err
+	}
+	if err := idtools.MkdirAllAs(filepath.Dir(d.block(id)), 0700, rootUID, rootGID); err != nil {
+		return err
+	}
+	if err := idtools.MkdirAllAs(d.mnt(id), 0755, rootUID, rootGID); err != nil {
+		return err
+	}
+	if parent == "" {
+		return d.createBaseBlock(id, mountLabel)
+	}
+	// TODO: use syscall directly, cp --reflink supports btrfs only currently
+	if out, err := exec.Command("cp", "-a", "--reflink=auto", d.block(parent), d.block(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to reflink:%v:%s", err, string(out))
+	}
+	return nil
+}
+
+func (d *Driver) block(id string) string {
+	return filepath.Join(d.home, "blocks", id)
+}
+
+func (d *Driver) mnt(id string) string {
+	return filepath.Join(d.home, "mnt", id)
+}
+
+func joinMountOptions(a, b string) string {
+	if a == "" {
+		return b
+	}
+	if b == "" {
+		return a
+	}
+	return a + "," + b
+}
+
+func (d *Driver) mount(id string, mountLabel string) error {
+	options := "loop"
+
+	if d.blockFs == "xfs" {
+		// XFS needs nouuid or it can't mount filesystems with the same fs
+		options = joinMountOptions(options, "nouuid")
+	}
+
+	options = joinMountOptions(options, label.FormatMountLabel("", mountLabel))
+
+	//if err := mount.Mount(d.block(id), d.mnt(id), d.blockFs, options); err != nil {
+	//	return fmt.Errorf("rawblock: Error mounting '%s' on '%s': %s", d.block(id), d.mnt(id), err)
+	//}
+	if out, err := exec.Command("mount", "-t", d.blockFs, "-o", options, d.block(id), d.mnt(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to mount block:%v:%s", err, string(out))
+	}
+	return nil
+}
+
+// Remove deletes the content from the directory for a given id.
+func (d *Driver) Remove(id string) error {
+	if err := os.RemoveAll(d.block(id)); err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if err := os.RemoveAll(d.mnt(id)); err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	return nil
+}
+
+// Get returns the directory for the given id.
+func (d *Driver) Get(id, mountLabel string) (string, error) {
+	d.Lock()
+	defer d.Unlock()
+	active := d.active[id]
```

Comment:
d.active does not seem reliable upon daemon restart. e.g., after daemon restarts, d.active[id] is zero and damon tries to mount the device again, which will fail if a device is mounted before.